src/compiler/parser.ts

```diff
@@ -17,19 +17,19 @@ namespace ts {
     }
 
     function visitNode<T>(cbNode: (node: Node) => T, node: Node): T {
-        if (node) {
+        if (node !== void 0) {

```

Comment:
I mentioned it in person, but how does this compare to an always `undefined` (never passed) function argument? (IE, `visitNode<T>(cbNode: (node: Node) => T, node: Node, missing?: undefined): T`).